main.tf

```diff
@@ -15,18 +15,28 @@
 */
 
 locals {
+  eni_configs = [for e in var.pods_subnets : {
+    name           = e.availability_zone
+    subnet         = e.id
+    securityGroups = [aws_eks_cluster.quortex.vpc_config[0].cluster_security_group_id]
+  }]
   # The Quortex cluster OIDC issuer.
   cluster_oidc_issuer = trimprefix(aws_eks_cluster.quortex.identity[0].oidc[0].issuer, "https://")
   node_group_labels = [
     for key, node_group in var.node_groups :
     {
-      for k, v in lookup(node_group, "labels", {}) :
+      for k, v in lookup(node_group, "labels", { "k8s.amazonaws.com/eniConfig" : data.aws_subnet.subnets[local.nodes_subnet_ids_map[key]].availability_zone }) :
       key => {
         "k8s.io/cluster-autoscaler/node-template/label/${k}" : v
         (k) : v
       }...
     }
   ]
+  nodes_subnet_ids_map = merge(
+    { for key, node_group in var.node_groups : key => lookup(node_group, "subnet_ids", [])[0] },
+    { for key, node_group in var.node_groups_advanced : key => lookup(node_group, "subnet_ids", [])[0] },
+  )
+  distinct_subnet_ids = distinct([for key, subnet in local.nodes_subnet_ids_map : subnet])
```

Comment:
A bit complicated for something that could have been written `toset(values(local.nodes_subnet_ids_map))`